app/cells/decidim/homepage_proposals/content_blocks/proposals_slider_cell.rb

```diff
@@ -54,6 +54,31 @@ def categories_filter
         def selected_component_id
           @selected_component_id ||= params.dig(:filter, :component_id) || content_block_settings.default_linked_component
         end
+
+        def scopes_filter
+          options = []
+          root = linked_components.collect(&:scope).collect(&:blank?).any?
+
+          if root
```

Comment:
I think it could be easier to read with something like : 

```suggestion
          scopes = if root
              current_organization.scopes.top_level
              else
              linked_components.collect(&:scope).uniq
          end
```